omc3/model/accelerators/lhc.py

```diff
@@ -486,12 +488,42 @@ def _get_madx_main_sequence_loading(self) -> str:
                 "has to be specified (--lhcmode option missing?)."
             )
 
-    def get_update_correction_script(self, outpath: Path | str, corr_files: Sequence[Path | str]) -> str:
+    def get_update_correction_script(self, outpath: Path | str, corr_files: Sequence[Path | str], variable_categories: Sequence[str]) -> str:
         madx_script = self.get_base_madx_script()
         for corr_file in corr_files:
             madx_script += f"call, file = '{str(corr_file)}';\n"
-        madx_script += f"exec, do_twiss_elements(LHCB{self.beam}, '{str(outpath)}', {self.dpp});\n"
+        
+        if DELTAP_NAME in variable_categories: # If the dpp is a variable in the corrections, do twiss, correct, match
+            madx_script += self.get_update_deltap_script()
+        else: # Else, just set the dpp and do twiss
+            madx_script += f"{DELTAP_NAME} = {self.dpp:.15e};\n"
+        madx_script += f"exec, do_twiss_elements(LHCB{self.beam}, '{str(outpath)}', {DELTAP_NAME});\n"
         return madx_script
+    
+    def get_update_deltap_script(self) -> str:
+        if self._uses_run3_macros():
+            knobx = f"dQx.b{self.beam}_op"
+            knoby = f"dQy.b{self.beam}_op"
+        elif int(self.year) == 2018: # This is very annoying - is there another way?
```

Comment:
Yes it is, but I don't think there is a better way. 
But I would move this whole block into its own function (e.g. `self.get_tune_knobs() -> tuple[str, str]:` and in this function `tune_knobs = self.get_tune_knobs()`, `"\n".join([f"vary, name={knob};" for knob in tune_knobs])`), as I easily imagine to have to use this again somewhere in the future.